specification/communication/resource-manager/Microsoft.Communication/preview/2022-10-01-preview/Domains.json

```diff
@@ -0,0 +1,1280 @@
+            "headers": {
+              "location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Cancel verification": {
+            "$ref": "./examples/domains/cancelVerification.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/listValidSenderUsernames": {
+      "post": {
+        "description": "Get a list of valid sender user names.",
+        "operationId": "Domains_ListValidSenderUsernames",
+        "summary": "List Valid Sender User Names",
+        "tags": [
+          "Domains"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "#/parameters/DomainName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes valid sender user names for the Domains resource.",
+            "schema": {
+              "$ref": "#/definitions/ValidSenderUsernameCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "List valid sender user names": {
+            "$ref": "./examples/domains/listValidSenderUsernames.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/addValidSenderUsernames": {
```

Comment:
>addValidSenderUsernames [](http://example.com/codeflow?start=160&length=23)

model this as a resource instead of all these POST actions.

Reply:
DomainsResource is an RPaaS tracked resource. Part of this change removes the ValidSenderUsername property from the Domains resource. The ValidSenderUsername is classifed as Customer Content EUII, and as such should not be cached/stored by RPaaS in the global datastores. RPaaS cached/stores resource data for all GET/PUT/PATCH requests.

Making these POST operations allows us to write to the correct geographic datastore for the customer content/euii data without storing any resource data in RPaaS.

This is similar to the RPaaS PureProxy feature, where services are allowed to use RPaaS as a passthrough, sending requests directly to the UserRP.

All these POST APIs are processing Customer Content and EUII.